src/helpers.c

```diff
@@ -102,17 +101,20 @@ int16_t bit_reverse(int16_t unreversed, int8_t bits_to_reverse) {
     return reversed;
 }
 
+
 long int tokentol_reverse(const char* str, char** endptr, int base) {
-    long int reverse_val;
-    reverse_val = strtol(strrev(str), endptr, base);
-    return reverse_val;
+    int16_t reverse_val;                                                                    
+    reverse_val = strtol(strrev(str), endptr, base);                                         
+    return reverse_val;                                                                      
 }
 
-const char* strrev(const char* str) {
-    int index = strlen(str);
-    char* revStr = (char*)malloc(index--);
-    int destIndex = 0;
-    while (0 <= index) revStr[destIndex++] = str[index--];
-    revStr[destIndex] = 0;
-    return revStr;
+const char* strrev(const char* str){
+    int8_t length = strlen(str);
+    char rev[length];
+    for(int8_t i = 0; i<length; i++){
+	rev[length-1-i] = str[i];
+    }
+    strcpy(str,rev);
```

Comment:
i think you need to include `\0` at the end of `rev` in order for `strcpy` to work properly. also, i'd make it return `void` since you're reverting the string passed as a parameter (and this also means it shouldn't be marked as `const` parameter).

i would just keep it simple, it's only used in one place and we know it's for a 16 char string, so you could just revert in place:

```
char c;
for (int8_t i = 0; i < 8; i++) {
    c = str[i];
    str[i] = str[15 - i];
    str[15 - i] = c;
}
```

Reply:
this was getting too convoluted anyhow, replaced everything with a single, simple rev_bitstring_to_int in helpers